### PR TITLE
[spark-3] Bump Apache spark to 3.0.0-preview2

### PR DESCRIPTION
**Pull request by yujiantao**

### CODE REVIEW

File: spark/src/test/java/org/apache/iceberg/spark/source/TestStructuredStreaming.java [hunk 1]
```diff
@@ -79,7 +80,7 @@ public static void stopSpark() {
   }
 
   @Test
-  public void testStreamingWriteAppendMode() throws IOException {
+  public void testStreamingWriteAppendMode() throws IOException, TimeoutException {
```
Comment (rdblue) [inline spark/src/test/java/org/apache/iceberg/spark/source/TestStructuredStreaming.java]: How about throwing `Exception`? Keeping a list of which checked exceptions may be thrown isn't useful for test methods.

File: spark/src/main/java/org/apache/iceberg/spark/source/SparkBatchWrite.java [hunk 2]
```diff
@@ -493,6 +494,9 @@ protected void setCurrentKey(PartitionKey currentKey) {
     public void write(InternalRow row) throws IOException {
       writeInternal(row);
     }
+
+    @Override
+    public void close() {}
```
Comment (rdblue) [inline spark/src/main/java/org/apache/iceberg/spark/source/SparkBatchWrite.java]: Can you add a comment to explain why this is empty? If I understand correctly, close is called after `abort` or `commit` is called. Both `abort` and `commit` will close the current data file so there is no more cleanup to do here.

**Comment by rdblue** [comment]
Thanks for working on this, @yujiantao! Could you fix the test errors? I also had a couple of minor comments.

File: spark/src/test/java/org/apache/iceberg/spark/source/TestStructuredStreaming.java (hunk 1 above)
Comment (yujiantao) [reply spark/src/test/java/org/apache/iceberg/spark/source/TestStructuredStreaming.java]: Yes, throwing "Exception" is better.

File: spark/src/main/java/org/apache/iceberg/spark/source/SparkBatchWrite.java (hunk 2 above)
Comment (yujiantao) [reply spark/src/main/java/org/apache/iceberg/spark/source/SparkBatchWrite.java]: I will add this comment.

**Comment by yujiantao** [comment]
@rdblue Sorry for late reply as I was on a long vocation. I have refreshed the commit.

File: spark/src/test/java/org/apache/iceberg/spark/source/TestTables.java [hunk 3]
```diff
@@ -37,7 +37,7 @@
 import org.apache.iceberg.io.InputFile;
 import org.apache.iceberg.io.LocationProvider;
 import org.apache.iceberg.io.OutputFile;
-import parquet.Preconditions;
+import org.apache.parquet.Preconditions;
```
Comment (rdblue) [inline spark/src/test/java/org/apache/iceberg/spark/source/TestTables.java]: This should not be using Preconditions from Parquet. It should be the guava one. Can you update it to that one instead?

File: spark/src/test/java/org/apache/iceberg/spark/source/TestForwardCompatibility.java [hunk 4]
```diff
@@ -125,7 +126,7 @@ public void testSparkWriteFailsUnknownTransform() throws IOException {
   }
 
   @Test
-  public void testSparkStreamingWriteFailsUnknownTransform() throws IOException {
+  public void testSparkStreamingWriteFailsUnknownTransform() throws IOException, TimeoutException {
```
Comment (rdblue) [inline spark/src/test/java/org/apache/iceberg/spark/source/TestForwardCompatibility.java]: This should also be `Exception`.

**Comment by rdblue** [comment]
Thanks @yujiantao! I caught a couple more minor things. Once those are done, we should be good to merge.

File: spark/src/test/java/org/apache/iceberg/spark/source/TestForwardCompatibility.java (hunk 4 above)
Comment (yujiantao) [reply spark/src/test/java/org/apache/iceberg/spark/source/TestForwardCompatibility.java]: Ok

File: spark/src/test/java/org/apache/iceberg/spark/source/TestTables.java (hunk 3 above)
Comment (yujiantao) [reply spark/src/test/java/org/apache/iceberg/spark/source/TestTables.java]: Yes.

**Comment by jerryshao** [comment]
LGTM.

**Comment by rdblue** [comment]
Looks good! Thanks for working on this @yujiantao!

**Pull request merged**